### PR TITLE
No longer auto-installing maistra on openshift [SRVKS-196]

### PR DESCRIPTION
**Pull request by jcrossley3**
I removed the SCC stuff, too, which I think was only required for maistra, but I need @nak3 to confirm

### CODE REVIEW

**Comment by nak3** [comment]
Please let me confirm. The SCC stuff is necessary for not only Maistra but also any Istio env when it uses side-car injection due to [requirement of NET_ADMIN](https://istio.io/docs/ops/setup/required-pod-capabilities/). And Knative's autoscaler and activator still uses the side-car injection as https://github.com/knative/serving/pull/966.

So, do you think that we should let users configure the SCC by themselves?

**Comment by markusthoemmes** [comment]
@nak3 SCC is not necessary as of Maistra TP12 (when using CNI). It's part of their multi-tenancy story to not need these cluster roles. The sidecars no longer need NET_ADMIN access as they don't setup the network through iptables anymore.

**Comment by nak3** [comment]
Thank you for the clarification. Then, I think that we do not need the `privileged`  SCC. 

But I think that the [`anyuid` SCC](https://github.com/openshift-knative/knative-serving-operator/pull/52/files#diff-65614fcbf6787b19ff4e6d1dc312c6ccL48) was added by @savitaashture. So  can you also confirm it?  @savitaashture

File: pkg/controller/knativeserving/knativeserving_controller.go [hunk 1]
```diff
@@ -228,6 +230,25 @@ func (r *ReconcileKnativeServing) checkDeployments(instance *servingv1alpha1.Kna
...
 
+// Check for all dependencies
+func (r *ReconcileKnativeServing) checkDependencies(instance *servingv1alpha1.KnativeServing) error {
+	defer r.updateStatus(instance)
+
+	istio := schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "virtualservice"}
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(istio)
+	if err := r.client.List(context.TODO(), nil, list); err != nil {
+		msg := fmt.Sprintf("Istio not detected")
```
Comment (markusthoemmes) [inline pkg/controller/knativeserving/knativeserving_controller.go]: In the interest of usability: Should we call out "ServiceMesh not detected" here? I know that technically we only need Istio, but the user of Openshift Serverless might think "hey, what the hell is Istio? I only see something like ServiceMesh referenced in the docs". Maybe we could also write something like "Istio not detected, ServiceMesh is likely not installed" to satisfy both ends?

**Comment by markusthoemmes** [comment]
@nak3 thanks for being on the watch 😂  The anyuid thing was needed for custom gateway and was indeed added by @savitaashture. Seems like we no longer need that in newer releases though as I haven't heard any complaints for this setup lately.

File: pkg/controller/knativeserving/knativeserving_controller.go (hunk 1 above)
Comment (jcrossley3) [reply pkg/controller/knativeserving/knativeserving_controller.go]: Good call. Done.

**Pull request merged**